src/analytics.ts

```diff
@@ -0,0 +1,74 @@
+import { AttributesTypes, Growthbook } from './growthbook'
+import { RudderStack, SignupProvider, TEvents } from './rudderstack'
+
+export type AnalyticsData = { event: keyof TEvents; form_source: string; form_name: string }
+type ExtractAction<T> = T extends { action: infer A } ? A : never
+type ActionForEvent<E extends keyof TEvents> = ExtractAction<TEvents[E]>
+type Options = {
+    growthbookKey: string
+    growthbookDecryptionKey: string
+    rudderstackKey: string
+    enableDevMode?: boolean
+}
+
+export function createAnalyticsInstance(options?: Options) {
+    let _growthbook: Growthbook, _rudderstack: RudderStack
+
+    const initialise = ({ growthbookKey, growthbookDecryptionKey, rudderstackKey, enableDevMode }: Options) => {
+        _rudderstack = RudderStack.getRudderStackInstance(rudderstackKey)
+        _growthbook = Growthbook.getGrowthBookInstance(
+            growthbookKey,
+            growthbookDecryptionKey,
+            enableDevMode,
+        )
+    }
+
+    if (options) {
+        initialise(options)
+    }
+
+    const setAttributes = ({ country, user_language, device_language, device_type }: AttributesTypes) => {
+        _growthbook.setAttributes({
+            id: getId(),
+            country,
+            user_language,
+            device_language,
+            device_type,
+        })
+    }
+
+    const getFeatureState = (id: string) => _growthbook.getFeatureState(id)?.experimentResult?.name
+    const getFeatureValue = (id: string) => _growthbook.getFeatureValue(id)
+    const getId = () => _rudderstack.getUserId() || _rudderstack.getAnonymousId()
+
+    const track = <T extends keyof TEvents>(
+        analyticsData: AnalyticsData,
+        action: ActionForEvent<T>,
+        signup_provider?: SignupProvider,
+    ) => {
+        _rudderstack.track(
+            analyticsData.event,
+            {
+                action,
+                signup_provider,
+                form_source: analyticsData.form_source,
+                form_name: analyticsData.form_name,
+            },
+            { is_anonymous: !!_rudderstack.getAnonymousId() },
```

Comment:
I think it's always going to be `true` because there will probably be anonymous id generated everytime when the instance is created.

Reply:
Yeah I think you're right. Is there any way to know if the user is logged in?

Reply:
We have to pass it due the strange `has_identified` value in rudderstack which block initialisation
TBH I want to remove these 2 params at all in the future. `is_anonymous` and `has_identified`

Reply:
Wrong name, but idea is clear
<img width="1077" alt="Screenshot 2023-10-12 at 09 18 56" src="https://github.com/binary-com/deriv-analytics/assets/103182473/62d73669-5998-4c2b-80cd-94e85174385b">

Reply:
we can use identifiedEvent to make has_indentified true but it requires call. Not clear what we getting